cmd/provisioner-localpv/app/config.go

```diff
@@ -0,0 +1,164 @@
+	if len(strings.TrimSpace(pvcCASConfigStr)) != 0 {
+		pvcCASConfig, err := cast.UnMarshallToConfig(pvcCASConfigStr)
+		if err == nil {
+			pvConfig = cast.MergeConfig(pvcCASConfig, pvConfig)
+		}
+	}
+
+	pvConfigMap, err := cast.ConfigToMap(pvConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read configuration for pvc %v", pvc.ObjectMeta.Name)
+	}
+
+	c := &CASConfigPVC{
+		pvName:  pvName,
+		pvcName: pvc.ObjectMeta.Name,
+		scName:  scName,
+		config:  pvConfigMap,
+	}
+	return c, nil
+}
+
+//GetPath returns a valid PV path based on the configuration
+// or an error. The Path is constructed using the following rules:
+// If AbsolutePath is specified return it. (Future)
+// If PVPath is specified, suffix it with BasePath and return it. (Future)
+// If neither of above are specified, suffix the PVName to BasePath
+//  and return it
+// Also before returning the path, validate that path is safe
+//  and matches the filters specified in StorageClass.
+func (c *CASConfigPVC) GetPath() (string, error) {
+	//This feature need to be supported with some more
+	// security checks are in place, so that rouge pods
+	// don't get access to node directories.
+	//absolutePath := c.getConfigValue(KeyPVAbsolutePath)
+	//if len(strings.TrimSpace(absolutePath)) != 0 {
+	//	return c.validatePath(absolutePath)
+	//}
+
+	basePath := c.getConfigValue(KeyPVBasePath)
+	if len(strings.TrimSpace(basePath)) == 0 {
+		return "", fmt.Errorf("configuration error, no base path was specified")
+	}
+
+	//This feature need to be supported after the
+	// security checks are in place.
+	//pvRelPath := c.getConfigValue(KeyPVRelativePath)
+	//if len(strings.TrimSpace(pvRelPath)) == 0 {
+	//	pvRelPath = c.pvName
+	//}
+
+	pvRelPath := c.pvName
+	path := filepath.Join(basePath, pvRelPath)
+
+	return c.validatePath(path)
+}
+
+//getConfigValue is a utility function to extract the value
+// of the `key` from the ConfigMap object - which is
+// map[string]interface{map[string][string]}
+func (c *CASConfigPVC) getConfigValue(key string) string {
+	if configObj, ok := util.GetNestedField(c.config, key).(map[string]string); ok {
+		if val, p := configObj[string(mconfig.ValuePTP)]; p {
+			return val
+		}
+	}
+	return ""
+}
+
+//validatePath checks for the sanity of the PV path
+func (c *CASConfigPVC) validatePath(path string) (string, error) {
+	//Validate that path is well formed.
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return "", err
+	}
+
+	//Validate that root directories are not used as PV paths
+	//Valid path should have a basepath (which is not /) and
+	// a subpath for the PV.
+	path = strings.TrimSuffix(path, "/")
+	parentDir, volumeDir := c.extractSubPath(path)
+	//parentDir = strings.TrimSuffix(parentDir, "/")
+	//volumeDir = strings.TrimSuffix(volumeDir, "/")
+	if parentDir == "" || volumeDir == "" {
+		// it covers the `/` case
+		return "", fmt.Errorf("invalid path %v for cleanup: cannot find parent dir or volume dir", path)
```

Comment:
```go
import "github.com/openebs/maya/pkg/errors"

return "", errors.Errorf(
  "validate path failed: invalid path {%v}: cannot find parent dir or volume dir", path
)
```

- We cannot use cleanup here, since function name talk about `validation`
- Need to use errors instead of fmt

Reply:
done.